File: Spark GraphX/graph info.scala
import org.apache.spark.sql.SparkSession
// To make some of the examples work we will also need RDD
import org.apache.spark.rdd.RDD
import org.apache.spark.graphx.GraphLoader
import org.apache.spark.sql.functions._
object ReadTextFile {

    def main() : Unit = {
        val spark:SparkSession = SparkSession.builder()
            .master("local[1]")
            .appName("Spark SQL")
            .getOrCreate()   

        val graph = GraphLoader.edgeListFile(sc, "file:///home/ccw/hadoop-2.10.1/input/edge_list.txt")
        println("Vertices count: " + graph.vertices.count)
        println("Edges count: " + graph.edges.count)
        println("inDegrees:")
        graph.inDegrees.sortBy(_._2, ascending=false).take(10).foreach(println)
        println("outDegrees")
        graph.outDegrees.sortBy(_._2, ascending=false).take(10).foreach(println)
    }
}

File: Spark GraphFrames/graph info.py
import os
from pyspark import SparkContext
from pyspark.sql import SQLContext
from pyspark.sql import SparkSession
from graphframes import *
if __name__ == "__main__":
    spark = SparkSession \
        .builder \
        .appName("Spark SQL") \
        .getOrCreate()
    spark.sparkContext.setLogLevel("ERROR")
    e = spark.read.csv("file:///home/ccw/hadoop-2.10.1/input/mooc_actions.tsv", sep=r'\t', header=True, inferSchema='true')
    v = spark.read.csv("file:///home/ccw/hadoop-2.10.1/input/vertices.tsv", sep=r'\t', header=True, inferSchema='true')
    g = GraphFrame(v, e)
    print("V count is " ,g.vertices.count())#count vertices
    print("Edge count is " , g.edges.count())#count edges
    print("Indegree: ")
    g.inDegrees.orderBy('inDegree', ascending=False).show(10)
    print("outdegree: ")
    g.outDegrees.orderBy('outDegree', ascending=False).show(10)

    g1 = g.filterEdges("TIMESTAMP >= 10000 and TIMESTAMP<=50000").dropIsolatedVertices()
    print("g1 Edge count is " , g1.edges.count())
    print("g1 vertices count is " , g1.vertices.count())